### ISSUE TITLE
[Remix + Vite] Trying to build production app failing (mixed client and server code)

### ISSUE THREAD

**Issue by robipop22** [vercel/remix#97]
This app is based on the https://github.com/robipop22/dnb-stack. The upgrade on the stack worked. 
Followed the same steps for my app but the build is failing even locally with :

` './entry.server.js' imported by 'node_modules/@vercel/remix/dist/edge/index.js'
`

Tried with package 2.8.0, 2.8.1, 2.8.1-patch2.

I commented in node modules the import : var entry_server = require('./entry.server.js');
it worked after.
But when I added back again the build started to fail again.

Any guidance or help would be much appreciated.

thanks

**Comment by TooTallNate** [comment]
What is the full error you are seeing? Can you share your code?

**Comment by robipop22** [comment]
`Error: [commonjs--resolver] Server-only module referenced by client

    './entry.server.js' imported by 'node_modules/@vercel/remix/dist/edge/index.js'

  See https://remix.run/docs/en/main/future/vite#splitting-up-client-and-server-code

file: ../node_modules/@vercel/remix/dist/edge/index.js`

Happy to share any code that you think is helpful, or jump on a discord call or something similar.

**Comment by TooTallNate** [comment]
If you can share the code that would be the most helpful. Also note that [`entry.server`](https://remix.run/docs/en/main/file-conventions/entry.server) is a special file name in Remix, so if you are trying to import it within one of your routes then it's probably improper usage.

**Comment by robipop22** [comment]
I'm using the defaults entry files. I deleted them so I make sure they won't interfere.

Tried a lot of workarounds, different vite configs. 
different package managers.

Fails with the same node modules error everytime

**Comment by TooTallNate** [comment]
I mean, it's something in your app code. It's not related to Vite config nor package manager nor even Vercel. You can try working backwards from the [starter template](https://github.com/vercel/vercel/tree/main/examples/remix) which does not exhibit this error. I'd be happy to help diagnose further if you want to share a reproduction.

**Comment by robipop22** [comment]
Feels like it, deleted the whole routes files then it failed again, then commented this file 
../node_modules/@remix-run/dev/dist/config/defaults/entry.client.tsx
it worked after, so it seems that one of the route might be at fault. trying to find out which one

**Comment by robipop22** [comment]
Well that was fun to debug, but in the end I found it. I'm going to leave here my findings in case someone else comes across this.

I had a .ts route that was handling a POST and GET so it was exposing an **action** and **loader**. 
But inside my action function I was calling a function (let's call it _myFunction_ ) 
And **myFunction** was doing in the end something like :
`return json({})`

It seems that since the function was inside the same file it got treated like client code.

By moving **myFunction** to another file called **myFunction.server.ts** and importing it into my route with the action it all got treated on the server and the build is working now.

I tried to explain it the best I could. I'll close this. Thanks @TooTallNate  for the quick response.
